lib/dry/view.rb

```diff
@@ -469,14 +467,20 @@ def call(format: config.default_format, context: config.default_context, **input
 
       if layout?
         layout_env = self.class.layout_env(format: format, context: context)
-        output = layout_env.template(self.class.layout_path, layout_env.scope(config.scope, layout_locals(locals))) { output }
+        output = env.template(self.class.layout_path, layout_env.scope(config.scope, layout_locals(locals))) { output }
       end
 
       Rendered.new(output: output, locals: locals)
     end
 
     private
 
+    # @api private
+    def ensure_config
+      raise UndefinedPathsError.new unless Array(config.paths).any?
+      raise UndefinedTemplateError.new unless config.template
```

Comment:
@solnic @flash-gordon Style question for you. Which would you consider better:

- Different error classes for each bit of config that is critical to rendering, like we have here, or
- A single `InvalidConfig` error that we raise with different messages based on which piece of config is invalid?

Reply:
@jodosha Actually, this might be good to ask you, too. Would you rather have a single class for config-related errors, or an individual class for each specific config key that could be invalid?

Reply:
@timriley No strong preference here. As a general rule that I follow, it's worth to have specific error objects, if you need to enrich them with debug information.

---

For instance the error `UndefinedPathsError` can be useful if raised with configured paths:

```ruby
raise UndefinedPathsError.new(config.paths) unless Array(config.paths).any?
```

So devs will see an informative error message:

```log
Dry::View::UndefinedPathsError: cannot find paths: `path/to/missing/directory'
```

Reply:
@jodosha Thanks for the feedback :) In this case I decided to unify the errors and pass through the missing configuration key as the extra data: 

```
      raise UndefinedConfigError.new(:paths) unless Array(config.paths).any?
      raise UndefinedConfigError.new(:template) unless config.template
```